ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -1,21 +1,52 @@
 #include "../PrjFSKext/public/PrjFSLogClientShared.h"
 #include "../PrjFSKext/public/PrjFSVnodeCacheHealth.h"
+#include "../PrjFSLib/Json/JsonWriter.hpp"
 #include "../PrjFSLib/PrjFSUser.hpp"
 #include <iostream>
+#include <mutex>
+#include <sstream>
 #include <OS/log.h>
 #include <IOKit/IOKitLib.h>
 #include <signal.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+using std::hex;
+using std::lock_guard;
+using std::mutex;
+using std::ostringstream;
+using std::string;
 
 static const char PrjFSKextLogDaemon_OSLogSubsystem[] = "org.vfsforgit.prjfs.PrjFSKextLogDaemon";
+static const int INVALID_SOCKET_FD = -1;
 
 static os_log_t s_daemonLogger, s_kextLogger;
 static IONotificationPortRef s_notificationPort;
 
-static void StartLoggingKextMessages(io_connect_t connection, io_service_t service, os_log_t daemonLogger, os_log_t kextLogger);
+static int s_messageListenerSocket = INVALID_SOCKET_FD;
+static const string MessageListenerSocketPath = "/usr/local/GitService/pipe/vfs-c780ac06-135a-4e9e-ab6c-d41e2d265baa";
+static const string InfoMessageEventName = "info";
```

Comment:
Is there a nice way to group the Info/Error/Heath together in C++?

It looks like from C++ 17 the below is valid:
struct EventName
{
    inline static const string Info= "info";
    inline static const string Error= "error";
    etc...
}

What we have certainly isn't a killer but it would be great if a more specific grouping is available.

Reply:
Sure we can group these

Reply:
@jeschu1 let me know what you think of the `enum` I added here to get some grouping of values.